controller.py

```diff
@@ -469,16 +478,30 @@ def _make_request(url, on_404, on_timeout, on_exception, allow_401=False):
                 auth_url, problem
             )
             return problem
+
+        for rel, candidates in hyperlinks_to_create:
+            hyperlink, is_modified = library.set_hyperlink(rel, *candidates)
+            if is_modified:
+                # We need to send an email to this email address about
+                # what just happened. This is either so the receipient
+                # can confirm that the address works, or to inform
+                # them a new library is using their address.
+                hyperlink.notify(self.emailer, self.app.url_for)
                     
-        catalog = OPDSCatalog.library_catalog(library)
+        # Create an OPDS 2 catalog containing all available
+        # information about the library.
+        catalog = OPDSCatalog.library_catalog(
+            library, include_private_information=True
+        )
 
+        # Annotate the catalog with some information specific to
+        # the transaction that's happening right now.
         public_key = auth_document.public_key
         if public_key and public_key.get("type") == "RSA":
             public_key = RSA.importKey(public_key.get("value"))
             encryptor = PKCS1_OAEP.new(public_key)
 
             if not library.short_name:
-                # TODO: Generate a short name based on the library's service area.
```

Comment:
This was a nice idea but I don't think it's worth the extra effort, since people commonly confuse this short name with the "short name" that's used in the circulation manager as the URL for designating a specific library.